include/kf/FltResource.h

```diff
@@ -33,6 +33,33 @@ namespace kf
             ::FltReleaseResource(*this);
         }
 
+        void lock()
+        {
+            KeEnterCriticalRegion();
```

Comment:
There is no need in `KeEnterCriticalRegion/KeLeaveCriticalRegion` if `FltXXX` functions are used.